File: frontend/src/commands/environment-transfer-history.ts
import { invoke } from './index'

export const environment_transfer_history_query_id = async (environmentUuid: string): Promise<any> => {
    return await invoke('environment_transfer_history_query_id', { environmentUuid })
};

export const environment_transfer_history_query = async (pageNum: number, pageSize: number): Promise<any> => {
    return await invoke('environment_transfer_history_query', { pageNum, pageSize })
};

export const environment_transfer_history_batch_create = async (
    environmentUuids: Array<String>,
    currentUserEmail: String,
): Promise<any> => {
    return await invoke('environment_transfer_history_batch_create', {
        environmentUuids,
        currentUserEmail,
    })
};

export const environment_transfer_history_delete = async (environmentUuid: String): Promise<any> => {
    return await invoke(' environment_transfer_history_delete', { environmentUuid })
};

File: crates/lp-views/src/command/environment_transfer_history.rs
use lp_cores::requests::JsonRespnse;

#[tauri::command]
pub async fn environment_transfer_history_query_id(
    environment_uuid: String,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(
        lp_services_remote::requests::environment_transfer_history::query_by_uuid(
            &environment_uuid,
        )
        .await?,
    )
}

#[tauri::command]
pub async fn environment_transfer_history_query(
    page_num: u32,
    page_size: u32,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(
        lp_services_remote::requests::environment_transfer_history::query(page_num, page_size)
            .await?,
    )
}

#[tauri::command]
pub async fn environment_transfer_history_batch_create(
    current_user_email: String,
    environment_uuids: Vec<String>,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(
        lp_services_remote::requests::environment_transfer_history::batch_create(
            environment_uuids,
            &current_user_email,
        )
        .await?,
    )
}

#[tauri::command]
pub async fn environment_transfer_history_delete(
    environment_uuid: String,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(
        lp_services_remote::requests::environment_transfer_history::delete_by_uuid(
            &environment_uuid,
        )
        .await?,
    )
}
